src/app/components/UserAssets/Vesting/useListOfUserVestings.ts

```diff
@@ -1,121 +1,91 @@
-import { ContractName } from 'utils/types/contracts';
 import { Asset } from 'types/asset';
-import { useAccount } from '../../../hooks/useAccount';
+import { useAccount } from 'app/hooks/useAccount';
 import { useEffect, useState } from 'react';
-import { ethGenesisAddress } from '../../../../utils/classifiers';
-import { bridgeNetwork } from '../../../pages/BridgeDepositPage/utils/bridge-network';
-import { getContract } from '../../../../utils/blockchain/contract-helpers';
-import { Chain } from '../../../../types';
-
-import vestingRegistryAbi from 'utils/blockchain/abi/VestingRegistry.json';
-
-type RegistryMethodTypes = 'getVesting' | 'getTeamVesting';
-
-export type Vesting = {
-  asset: Asset;
-  label: string;
-  registry: ContractName;
-  registryMethod: RegistryMethodTypes;
-  staking: ContractName;
-};
-
-type StakesProp = {
-  dates: Date[];
-  stakes: string[];
-};
-
-export type FullVesting = Vesting & {
-  vestingContract: string;
-  balance: string;
-  stakes: StakesProp;
-};
-
-const possibleVestings: Vesting[] = [
-  {
-    asset: Asset.SOV,
-    label: 'Genesis SOV',
-    registry: 'vestingRegistry',
-    registryMethod: 'getVesting',
-    staking: 'staking',
-  },
-  {
-    asset: Asset.SOV,
-    label: 'Reward SOV',
-    registry: 'vestingRegistryLM',
-    registryMethod: 'getVesting',
-    staking: 'staking',
-  },
-  {
-    asset: Asset.SOV,
-    label: 'Team SOV',
-    registry: 'vestingRegistry',
-    registryMethod: 'getTeamVesting',
-    staking: 'staking',
-  },
-  {
-    asset: Asset.SOV,
-    label: 'Origin SOV',
-    registry: 'vestingRegistryOrigin',
-    registryMethod: 'getVesting',
-    staking: 'staking',
-  },
-  {
-    asset: Asset.FISH,
-    label: 'Origin FISH',
-    registry: 'vestingRegistryFISH',
-    registryMethod: 'getVesting',
-    staking: 'FISH_staking',
-  },
-  {
-    asset: Asset.FISH,
-    label: 'Airdrop FISH',
-    registry: 'vestingRegistryFISH',
-    registryMethod: 'getTeamVesting',
-    staking: 'FISH_staking',
-  },
-];
+import { ethGenesisAddress } from 'utils/classifiers';
+import { Vesting, FullVesting, DetailsProps } from './types';
+import { useVesting_getVestingsOf } from 'app/hooks/staking/useVesting_getVestingsOf';
+import { contractReader } from 'utils/sovryn/contract-reader';
 
 export function useListOfUserVestings(asset?: Asset) {
   const account = useAccount();
-
   const [items, setItems] = useState<FullVesting[]>([]);
   const [loading, setLoading] = useState(false);
+  const {
+    loading: loadingVestings,
+    value: vestingsContracts,
+  } = useVesting_getVestingsOf(account);
+  const [possibleVestings, setPossibleVestings] = useState<Vesting[]>([]);
+
+  useEffect(() => {
+    async function getVestings() {
+      try {
+        const address: string[] = [],
+          type: string[] = [],
+          typeCreation: string[] = [];
```

Comment:
Please don't use chained declarations.

```suggestion
        const address: string[] = [];
        const type: string[] = [];
        const typeCreation: string[] = [];
```